qa/tasks/cephadm.py

```diff
@@ -447,6 +447,70 @@ def pull_image(ctx, config):
     finally:
         pass
 
+@contextlib.contextmanager
+def setup_ca_signed_keys(ctx, config):
+    # generate our ca key
+    cluster_name = config['cluster']
+    bootstrap_remote = ctx.ceph[cluster_name].bootstrap_remote
+    bootstrap_remote.run(args=[
+        'sudo', 'ssh-keygen', '-t', 'rsa', '-f', '/root/ca-key', '-N', '""'
+    ])
+
+    # not using read_file here because it runs dd as a non-root
+    # user and would hit permission issues
+    r = bootstrap_remote.run(args=[
+        'sudo', 'cat', '/root/ca-key.pub'
+    ], stdout=StringIO())
+    ca_key_pub_contents = r.stdout.getvalue()
+
+    # make CA key accepted on each host
+    for remote in ctx.cluster.remotes.keys():
+        # write key to each host's /etc/ssh dir
+        remote.run(args=[
+            'sudo', 'echo', ca_key_pub_contents,
+            run.Raw('|'),
+            'sudo', 'tee', '-a', '/etc/ssh/ca-key.pub',
+        ])
+        # make sshd accept the CA signed key
+        remote.run(args=[
+            'sudo', 'echo', 'TrustedUserCAKeys /etc/ssh/ca-key.pub',
+            run.Raw('|'),
+            'sudo', 'tee', '-a', '/etc/ssh/sshd_config',
+            run.Raw('&&'),
+            'sudo', 'systemctl', 'restart', 'sshd',
+        ])
+
+    # generate a new key pair and sign the pub key to make a cert
+    bootstrap_remote.run(args=[
+        'sudo', 'ssh-keygen', '-t', 'rsa', '-f', '/root/cephadm-ssh-key', '-N', '""',
+        run.Raw('&&'),
+        'sudo', 'ssh-keygen', '-s', '/root/ca-key', '-I', 'user_root', '-n', 'root', '-V', '+52w', '/root/cephadm-ssh-key',
```

Comment:
```
2023-07-12T12:57:31.361 DEBUG:teuthology.orchestra.run.smithi033:> sudo ssh-keygen -t rsa -f /root/cephadm-ssh-key -N '""' && sudo ssh-keygen -s /root/ca-key -I user_root -n root -V +52w /root/cephadm-ssh-key
2023-07-12T12:57:32.069 INFO:teuthology.orchestra.run.smithi033.stdout:Generating public/private rsa key pair.
2023-07-12T12:57:32.070 INFO:teuthology.orchestra.run.smithi033.stdout:Your identification has been saved in /root/cephadm-ssh-key.
2023-07-12T12:57:32.070 INFO:teuthology.orchestra.run.smithi033.stdout:Your public key has been saved in /root/cephadm-ssh-key.pub.
2023-07-12T12:57:32.070 INFO:teuthology.orchestra.run.smithi033.stdout:The key fingerprint is:
2023-07-12T12:57:32.070 INFO:teuthology.orchestra.run.smithi033.stdout:SHA256:5Tpac48YhMFyB0XNptY/VF7ftRag/mxIOwbEogkjfrc root@smithi033
2023-07-12T12:57:32.070 INFO:teuthology.orchestra.run.smithi033.stdout:The key's randomart image is:
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:+---[RSA 3072]----+
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:|      .ooo   ..  |
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:|     . .. + . ..o|
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:| . o. +..*.. o .*|
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:|. . ooo+=oo . .oo|
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:| . . +..S..=  .  |
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:|  . . .. .o B    |
2023-07-12T12:57:32.071 INFO:teuthology.orchestra.run.smithi033.stdout:|     E  * .= =   |
2023-07-12T12:57:32.072 INFO:teuthology.orchestra.run.smithi033.stdout:|       o *.oo    |
2023-07-12T12:57:32.072 INFO:teuthology.orchestra.run.smithi033.stdout:|      . . . .    |
2023-07-12T12:57:32.072 INFO:teuthology.orchestra.run.smithi033.stdout:+----[SHA256]-----+
2023-07-12T12:57:32.082 INFO:teuthology.orchestra.run.smithi033.stderr:Enter passphrase: Load key "/root/ca-key": incorrect passphrase supplied to decrypt private key
2023-07-12T12:57:32.083 DEBUG:teuthology.orchestra.run:got remote process result: 255
```
https://pulpito.ceph.com/adking-2023-07-12_12:36:02-orch:cephadm-wip-adk4-testing-2023-07-11-2321-distro-default-smithi/7335103/

This seems to be because the earlier commands to make keys (both here and for the ca-key) are passing `-N '""'` so the password on the key ends up as `""` instead of being blank as intended.